VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
```

Comment:
Note that the documentation for `getExternalFilesDir()` states that it returns `null` if shared storage is not available, while the `getExternalStorageDirectory()` documentation states that it will return the `File` representing the directory path regardless of whether it's currently accessible.

Possible fixes:

1. Revert back to the old handcrafted path.

2. If the external application directory isn't available, automatically fall back to saving and fetching from the internal storage instead. Maybe move the data across when the external media is mounted.

2. Change the calling code to check for `null` and abort the download with an error message in that case, or prompt the user to select a valid download location.

Option 1 seems to be the simplest, especially keeping in mind the fact that this change is technically out of scope here.